src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Properties/InterceptedProjectProperties/BuildPropertyPage/TrimmingEnumProvider.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using Microsoft.Build.Framework.XamlTypes;
+using Microsoft.VisualStudio.ProjectSystem.Properties;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Properties;
+
+[ExportDynamicEnumValuesProvider("TrimmingEnumProvider")]
+[AppliesTo(ProjectCapability.DotNet)]
+internal class TrimmingEnumProvider : IDynamicEnumValuesProvider
+{
+    public Task<IDynamicEnumValuesGenerator> GetProviderAsync(IList<NameValuePair>? options)
+    {
+        return Task.FromResult<IDynamicEnumValuesGenerator>(new TrimmingEnumGenerator());
+    }
+
+    private class TrimmingEnumGenerator : IDynamicEnumValuesGenerator
```

Comment:
sealed class

Reply:
This is already restricting access to the generator class within its containing class, the provider class; so, my sense is that it's sufficient marking it as private.

Reply:
https://github.com/dotnet/project-system/pull/9140
This PR may start causing warnings and build failures. But you can take whichever direction you like.